moirae/models/ecm/__init__.py

```diff
@@ -10,18 +10,31 @@
 from .utils import hysteresis_solver_const_sign
 
 
+# TODO (wardlt): Does "constant" mean the previous or current value is used in span between last timestep and current.
 class EquivalentCircuitModel(CellModel):
     """
     Equivalent Circuit Model (ECM) dynamics of a battery
+
+    The only option for how to implement the battery is whether we assume the current
+    varies linearly between the previous and current step, or whether it is assumed to be constant.
+
+    Args:
+        current_behavior: How the current is assumed to vary between timesteps.
     """
 
-    @staticmethod
-    def update_transient_state(new_input: ECMInput,
-                               transient_state: ECMTransientVector,
-                               asoh: ECMASOH,
-                               previous_input: ECMInput,
-                               current_behavior: Literal['constant', 'linear'] = 'constant'
-                               ) -> ECMTransientVector:
+    current_behavior: str
+    """How it is assumed the current varies between timesteps"""
+
+    def __init__(self, current_behavior: Literal['constant', 'linear'] = 'constant'):
+        self.current_behavior = current_behavior
```

Comment:
I am opposed to this. I like the idea you had earlier of the `ECM` model being fully standalone, so that it always works in the same way given what is provided to it. If it's ok with you, I'd prefer just keeping the `current_behavior` as an argument to the main methods this class has, and with the `'constant'` default.

Reply:
I disagree that being able to change the current behavior is sufficient reason to deviate from the interface of CellModel. 

Two alternatives:
- If the current behavior is indeed something which can change between calls to the update and output function, should make it part of the control variables? 
- If it is something that all CellModels will use, should we make it as argument in the base CellModel class?

Reply:
Per my recollection of discussion, we're keeping it this way for now